lib/modules/login/login_interactor.dart

```diff
@@ -1,7 +1,27 @@
 part of 'login_module.dart';
 
-abstract class LoginInteractor extends Interactor<LoginInteractorDelegate> {}
+abstract class LoginInteractor extends Interactor<LoginInteractorDelegate> {
+  void login(String email, String password);
+}
 
-abstract class LoginInteractorDelegate {}
+abstract class LoginInteractorDelegate {
+  BehaviorSubject<void> get didLogin;
 
-class LoginInteractorImpl extends LoginInteractor {}
+  BehaviorSubject<Exception> get didFailToLogin;
+}
+
+class LoginInteractorImpl extends LoginInteractor {
+  final AuthRepository _authRepository = locator.get();
+
+  @override
+  void login(String email, String password) {
+    Future.microtask(() async {
```

Comment:
this is something new to me 🤔  but it seems like microtask() will take the priority highest; in a real project situation where there could be multiple request/async activities happening in the background, I think taking the queue like this could potentially create race condition issues, just saying we might have to be mindful about using it 🤔

Reply:
@sleepylee yea you're right, let me update it! 👍